service/storage_proxy.cc

```diff
@@ -4959,7 +4988,14 @@ void storage_proxy::init_messaging_service() {
                 opts.digest_algo = da;
                 opts.request = da == query::digest_algorithm::none ? query::result_request::only_result : query::result_request::result_and_digest;
                 auto timeout = t ? *t : db::no_timeout;
-                return p->query_result_local(std::move(s), cmd, std::move(pr2.first), opts, trace_state_ptr, timeout);
+                return p->query_result_local(std::move(s), cmd, std::move(pr2.first), opts, trace_state_ptr, timeout).handle_exception_type([&p] (const exceptions::overloaded_exception& e) {
+                    if (p->features().cluster_supports_query_status_for_reads()) {
+                        return make_ready_future<rpc::tuple<foreign_ptr<lw_shared_ptr<query::result>>, cache_temperature, query::status>>(
+                                rpc::tuple(make_foreign(make_lw_shared<query::result>()), cache_temperature::invalid(), query::status::OVERLOADED));
+                    } else {
```

Comment:
I still don't understand why you can't do this - return "status" unconditionally - without checking any flag. Isn't this the whole point of the IDL, that we can add new elements in structures and old nodes will just ignore them?

Reply:
https://github.com/scylladb/scylla/pull/7773#discussion_r539194347